scripts/test.sh

```diff
@@ -81,6 +89,7 @@ main() {
             then chapter=4 ; test_source_framework $s $i $chapter $variant
```

Comment:
use_source_test=$(awk 'FNR==1{ if ($0~"source-test") print "yes" }' $i)

this is a confusing choice. The test framework is called "framework", so it should be

       use_source_test=$(awk 'FNR==1{ if ($0~"framework") print "yes" }' $i)